### PR TITLE
fix the tf bug

### PR DESCRIPTION
**Pull request by SunAhong1993**
TensorFlow修复：
1. 修复Equal等bool输出类型的op，输出类型不正确问题。
2. 修复模型输入类型不正确问题。
3. 修复多输入混淆问题。
4. 修复elementwise类op有axis的问题。
transpose修改前后对比：
（第一行：前——release0.9上测试 ；第二行：后——develop上测试）（由于develop加了prelu的优化，所以有一些最初transpose的个数多一些）
EfficientNet
Transpose layers optimized, before: transpose_num=388, after: transpose_num=2
Transpose layers optimized, before: transpose_num=620, after: transpose_num=2
MTCNN-RNet
Transpose layers optimized, before: transpose_num=10, after: transpose_num=2
Transpose layers optimized, before: transpose_num=16, after: transpose_num=2
MobileNetV1
Transpose layers optimized, before: transpose_num=112, after: transpose_num=2
Transpose layers optimized, before: transpose_num=112, after: transpose_num=2
KerasBert
Transpose layers optimized, before: transpose_num=144, after: transpose_num=144
Transpose layers optimized, before: transpose_num=144, after: transpose_num=144
MTCNN-ONet
Transpose layers optimized, before: transpose_num=14, after: transpose_num=2
Transpose layers optimized, before: transpose_num=22, after: transpose_num=2
ResNetV2
Transpose layers optimized, before: transpose_num=426, after: transpose_num=2
Transpose layers optimized, before: transpose_num=426, after: transpose_num=2
UNet-GanUNet
Transpose layers optimized, before: transpose_num=32, after: transpose_num=32
Transpose layers optimized, before: transpose_num=32, after: transpose_num=32
YOLOv3
Transpose layers optimized, before: transpose_num=308, after: transpose_num=8
Transpose layers optimized, before: transpose_num=308, after: transpose_num=8
InceptionV4
Transpose layers optimized, before: transpose_num=634, after: transpose_num=2
Transpose layers optimized, before: transpose_num=634, after: transpose_num=2
MTCNN-PNet
Transpose layers optimized, before: transpose_num=14, after: transpose_num=4
Transpose layers optimized, before: transpose_num=20, after: transpose_num=4
MobileNetV2
Transpose layers optimized, before: transpose_num=142, after: transpose_num=2
Transpose layers optimized, before: transpose_num=142, after: transpose_num=2
ShuffleNet
Transpose layers optimized, before: transpose_num=239, after: transpose_num=52
Transpose layers optimized, before: transpose_num=239, after: transpose_num=52
UNet-UNet
Transpose layers optimized, before: transpose_num=64, after: transpose_num=26
Transpose layers optimized, before: transpose_num=64, after: transpose_num=26
AlBert
Transpose layers optimized, before: transpose_num=96, after: transpose_num=96
Transpose layers optimized, before: transpose_num=96, after: transpose_num=96
SqueezeNet
Transpose layers optimized, before: transpose_num=60, after: transpose_num=2
Transpose layers optimized, before: transpose_num=60, after: transpose_num=2
ResNetV1
Transpose layers optimized, before: transpose_num=434, after: transpose_num=2
Transpose layers optimized, before: transpose_num=434, after: transpose_num=2
MNASNetA1
Transpose layers optimized, before: transpose_num=228, after: transpose_num=2
Transpose layers optimized, before: transpose_num=228, after: transpose_num=2
VGG16
Transpose layers optimized, before: transpose_num=42, after: transpose_num=2
Transpose layers optimized, before: transpose_num=42, after: transpose_num=2
InceptionResNetV2
Transpose layers optimized, before: transpose_num=908, after: transpose_num=2
Transpose layers optimized, before: transpose_num=908, after: transpose_num=2
RetinaFace
Transpose layers optimized, before: transpose_num=210, after: transpose_num=28
Transpose layers optimized, before: transpose_num=210, after: transpose_num=28
ToyUNet-ToyUnet
Transpose layers optimized, before: transpose_num=30, after: transpose_num=26
Transpose layers optimized, before: transpose_num=30, after: transpose_num=26
ToyUNet-UNet
Transpose layers optimized, before: transpose_num=64, after: transpose_num=26
Transpose layers optimized, before: transpose_num=64, after: transpose_num=26

### CODE REVIEW

**Comment by jiangjiajun** [comment]
> TensorFlow修复：
> 
> 1. 修复Equal等bool输出类型的op，输出类型不正确问题。
> 2. 修复模型输入类型不正确问题。
> 3. 修复多输入混淆问题。
> 4. 修复elementwise类op有axis的问题。
>    transpose修改前后对比：
>    （第一行：前——release0.9上测试 ；第二行：后——develop上测试）（由于develop加了prelu的优化，所以有一些最初transpose的个数多一些）
>    EfficientNet
>    Transpose layers optimized, before: transpose_num=388, after: transpose_num=2
>    Transpose layers optimized, before: transpose_num=620, after: transpose_num=2
>    MTCNN-RNet
>    Transpose layers optimized, before: transpose_num=10, after: transpose_num=2
>    Transpose layers optimized, before: transpose_num=16, after: transpose_num=2
>    MobileNetV1
>    Transpose layers optimized, before: transpose_num=112, after: transpose_num=2
>    Transpose layers optimized, before: transpose_num=112, after: transpose_num=2
>    KerasBert
>    Transpose layers optimized, before: transpose_num=144, after: transpose_num=144
>    Transpose layers optimized, before: transpose_num=144, after: transpose_num=144
>    MTCNN-ONet
>    Transpose layers optimized, before: transpose_num=14, after: transpose_num=2
>    Transpose layers optimized, before: transpose_num=22, after: transpose_num=2
>    ResNetV2
>    Transpose layers optimized, before: transpose_num=426, after: transpose_num=2
>    Transpose layers optimized, before: transpose_num=426, after: transpose_num=2
>    UNet-GanUNet
>    Transpose layers optimized, before: transpose_num=32, after: transpose_num=32
>    Transpose layers optimized, before: transpose_num=32, after: transpose_num=32
>    YOLOv3
>    Transpose layers optimized, before: transpose_num=308, after: transpose_num=8
>    Transpose layers optimized, before: transpose_num=308, after: transpose_num=8
>    InceptionV4
>    Transpose layers optimized, before: transpose_num=634, after: transpose_num=2
>    Transpose layers optimized, before: transpose_num=634, after: transpose_num=2
>    MTCNN-PNet
>    Transpose layers optimized, before: transpose_num=14, after: transpose_num=4
>    Transpose layers optimized, before: transpose_num=20, after: transpose_num=4
>    MobileNetV2
>    Transpose layers optimized, before: transpose_num=142, after: transpose_num=2
>    Transpose layers optimized, before: transpose_num=142, after: transpose_num=2
>    ShuffleNet
>    Transpose layers optimized, before: transpose_num=239, after: transpose_num=52
>    Transpose layers optimized, before: transpose_num=239, after: transpose_num=52
>    UNet-UNet
>    Transpose layers optimized, before: transpose_num=64, after: transpose_num=26
>    Transpose layers optimized, before: transpose_num=64, after: transpose_num=26
>    AlBert
>    Transpose layers optimized, before: transpose_num=96, after: transpose_num=96
>    Transpose layers optimized, before: transpose_num=96, after: transpose_num=96
>    SqueezeNet
>    Transpose layers optimized, before: transpose_num=60, after: transpose_num=2
>    Transpose layers optimized, before: transpose_num=60, after: transpose_num=2
>    ResNetV1
>    Transpose layers optimized, before: transpose_num=434, after: transpose_num=2
>    Transpose layers optimized, before: transpose_num=434, after: transpose_num=2
>    MNASNetA1
>    Transpose layers optimized, before: transpose_num=228, after: transpose_num=2
>    Transpose layers optimized, before: transpose_num=228, after: transpose_num=2
>    VGG16
>    Transpose layers optimized, before: transpose_num=42, after: transpose_num=2
>    Transpose layers optimized, before: transpose_num=42, after: transpose_num=2
>    InceptionResNetV2
>    Transpose layers optimized, before: transpose_num=908, after: transpose_num=2
>    Transpose layers optimized, before: transpose_num=908, after: transpose_num=2
>    RetinaFace
>    Transpose layers optimized, before: transpose_num=210, after: transpose_num=28
>    Transpose layers optimized, before: transpose_num=210, after: transpose_num=28
>    ToyUNet-ToyUnet
>    Transpose layers optimized, before: transpose_num=30, after: transpose_num=26
>    Transpose layers optimized, before: transpose_num=30, after: transpose_num=26
>    ToyUNet-UNet
>    Transpose layers optimized, before: transpose_num=64, after: transpose_num=26
>    Transpose layers optimized, before: transpose_num=64, after: transpose_num=26

这里不是看这次PR后的效果，而是看这次PR对优化效果的影响
